### DISCUSSION TITLE
queryRasterElevation

### DISCUSSION THREAD

**Discussion by flemmens** [maplibre/maplibre-gl-js#3678] [Ideas]
We have a function (queryTerrainElevation) to retrieve the elevation of a location in Terrain mode, which is very usefull for instance to display the elevation under the cursor or query the elevation along a line.

```
map.on('mousemove', function(e) {
   console.log('Elevation under cursor is:', map.queryTerrainElevation(e.lngLat))
})
```

However, there is no equivalent to do this with a raster-dem when displaying a hillshade map. It could be very usefull as the purpose of hillshading is to show the relief and we don't necessarily want to activate the Terrain mode, which has a different look and is much more GPU heavy.

I would suggest to add a queryRasterElevation in the Camera class, just like queryTerrainElevation.

**Comment by HarelM** [comment]
I personally parse the tiles and generate elevation data from it in my project.
Note that terrain is global definition as opposed to raster-dem which is a source/hillshade layer.
It might be possible to add to the raster-dem source class a method to get elevation instead of the camera, IDK, feel free to pursue this.
Note that you can have multiple sources of terrain-rgb with different encoding/configurations, so it might be more "correct" to place it in the raster dem class.
In case you are interested, my code is here:
https://github.com/IsraelHikingMap/Site/blob/de4eff0ad97189fa7ebafc5be9a36aefd93dd595/IsraelHiking.Web/src/application/services/elevation.provider.ts#L105

**Reply by flemmens** [reply]
Thanks Harel, OK I will propose some update to the raster-dem class. Thank you for sharing your code as well.
